app/routers/event.py

```diff
@@ -40,32 +49,46 @@ def is_date_before(start_date: datetime, end_date: datetime) -> bool:
     return start_date < end_date
 
 
-def is_it_possible_to_change_dates(
+def is_change_dates_allowed(
         db: Session, old_event: Event, event: Dict[str, Any]) -> bool:
     return is_date_before(
         event.get('start', old_event.start),
         event.get('end', old_event.end))
 
 
-def get_items_that_can_be_updated(event: Dict[str, Any]) -> Dict[str, Any]:
-    """Extract only that keys to update"""
+def is_fields_types_valid(to_check: Dict[str, Any], types: Dict[str, Any]
+                          ) -> bool:
+    """validate dictionary values by dictionary of types"""
+    for key in to_check.keys():
+        if types[key] and not isinstance(to_check[key], types[key]):
+            # for future log
+            # print(key, 'is "'+type(to_check[key]).__name__+
+            #    '" and it should be from type "'+ types[key].__name__+ '"')
+            return False
+    return True
+
 
-    return {i: event[i] for i in (
-        'title', 'start', 'end', 'content', 'location') if i in event}
+def get_event_with_editable_fields_only(event: Dict[str, Any]
+                                        ) -> Dict[str, Any]:
+    """Remove all keys that are not allowed to update"""
+
+    return {i: event[i] for i in UPDATE_EVENTS_FIELDS.keys() if i in event}
 
 
 def update_event(event_id: int, event: Dict, db: Session
                  ) -> Optional[Event]:
 
     # TODO Check if the user is the owner of the event.
-
-    event_to_update = get_items_that_can_be_updated(event)
-    if not event_to_update:
+    event_to_update = get_event_with_editable_fields_only(event)
+    if not is_fields_types_valid(event_to_update, UPDATE_EVENTS_FIELDS):
         return None
     try:
         old_event = by_id(db=db, event_id=event_id)
-        if old_event is None or not is_it_possible_to_change_dates(
-                db, old_event, event_to_update):
+
+        if (not event_to_update
```

Comment:
Always prefer splitting to short lines, especially on indented code.
It can make great difference in the readability:

```python
forbidden = not is_change_dates_allowed(db, old_event, event_to_update)
if not event_to_update or not old_event is None or forbidden:
    return None
```